src/video_core/command_classes/codecs/codec.cpp

```diff
@@ -96,18 +209,15 @@ void Codec::Decode() {
         vp9_hidden_frame = vp9_decoder->WasFrameHidden();
     }
 
-    packet.data = frame_data.data();
-    packet.size = static_cast<s32>(frame_data.size());
-
-    avcodec_send_packet(av_codec_ctx, &packet);
-
-    if (!vp9_hidden_frame) {
-        // Only receive/store visible frames
-        AVFramePtr frame = AVFramePtr{av_frame_alloc(), AVFrameDeleter};
-        avcodec_receive_frame(av_codec_ctx, frame.get());
-        av_frames.push(std::move(frame));
-        // Limit queue to 10 frames. Workaround for ZLA decode and queue spam
-        if (av_frames.size() > 10) {
+    RawFrame raw_frame{
+        .frame_data = frame_data,
+        .vp9_hidden_frame = vp9_hidden_frame,
+    };
+    // TODO async
+    AVFrame* sw_frame = DecodeImpl(raw_frame);
+    if (sw_frame) {
+        if (av_frames.push(AVFramePtr{sw_frame, AVFrameDeleter}); av_frames.size() > 10) {
```

Comment:
```suggestion
        av_frames.push(AVFramePtr{sw_frame, AVFrameDeleter});
        if (av_frames.size() > 10) {
```
No benefit to having the queue push in the condition statement. Much easier to read breaking this up